subprojects/scala/src/integTest/groovy/org/gradle/scala/compile/UpToDateScalaCompileIntegrationTest.groovy

```diff
@@ -54,8 +54,8 @@ class UpToDateScalaCompileIntegrationTest extends AbstractIntegrationSpec {
 
         where:
         newScalaVersion | newZincVersion
-        '2.11.12'       | '1.2.0'
-        '2.12.6'        | '1.2.5'
+        '2.11.12'       | '1.6.0'
```

Comment:
It doesn't seem possible to use older Zinc versions. I don't even think this should be possible.

Reply:
If I understand correctly, this backward compatibility test is broken by your changes.
Compiling Scala 2.11 code is no longer possible, any other recent version _must_ upgrade zinc to 1.6.

In that case, this means this PR should wait until Gradle 8.0 because of that.

Am I right or is there some element I do not understand since I am not a Scala dev?

Reply:
> Compiling Scala 2.11 code is no longer possible, any other recent version must upgrade zinc to 1.6.

The test still passes, it's just not possible to set Zinc versions to be lower than 1.6.0, since we get binary compat issues (the method invoked is different between versions). However, to be honest it should never be needed to set the Zinc version manually to lower one, since you would be omitting a lot of bugfixes etc. without any possible gain. It's true though if someone does have it set manually it will not work anymore. It's probably possible to make it work via some kind of reflection, but that would require a lot more effort.

The other option would be to deprecate the zinc version setting and ignore it in the settings.

Reply:
@ljacomet why do we say that compiling with Scala 2.11 is not possible?

Reply:
I do not know - re-reading my comment and I am not sure what I meant exactly 😅 

What remains is the potential backwards compatibility issue. But I also understand that moving forward would be really beneficial to Scala users.
The window for getting this in Gradle 7.5 is still open. Let me check internally if we are OK with this.

Reply:
The only problem would be if users had the zinc version specified manually, though this should be a really rare scenario and something that should be discouraged. I is a potential compatibility issue though if someone had an old zinc version set.